--- doc/l2tpd.conf.sample.org	Sat Aug 10 02:42:22 2002
+++ doc/l2tpd.conf.sample	Sun Nov 10 11:52:46 2002
@@ -10,7 +10,7 @@
 ;
 ; [global]								; Global parameters:
 ; port = 1701						 	; * Bind to port 1701
-; auth file = /etc/l2tp/l2tp-secrets 	; * Where our challenge secrets are
+; auth file = /etc/l2tpd/l2tp-secrets 	; * Where our challenge secrets are
 ; access control = yes					; * Refuse connections without IP match
 ; rand source = dev                     ; Source for entropy for random
 ;                                       ; numbers, options are:
--- doc/l2tp-secrets.5.org	Mon Aug 19 19:14:34 2002
+++ doc/l2tp-secrets.5	Sun Nov 10 14:47:58 2002
@@ -27,7 +27,7 @@
 
 .SH "FILES"
 
-\fB\fR/etc/l2tp/l2tpd.conf \fB\fR/etc/l2tp/l2tp\-secrets 
+\fB\fR/etc/l2tpd/l2tpd.conf \fB\fR/etc/l2tpd/l2tp\-secrets 
 \fB\fR/var/run/l2tp\-control
 .SH "BUGS"
 
--- file.h.org	Wed Apr 10 22:09:33 2002
+++ file.h	Sun Nov 10 14:54:50 2002
@@ -49,10 +49,10 @@
 #define SENSE_ALLOW -1
 #define SENSE_DENY 0
 
-#define DEFAULT_AUTH_FILE "/etc/l2tp/l2tp-secrets"
-#define ALT_DEFAULT_AUTH_FILE "/etc/l2tpd/l2tp-secrets"
-#define DEFAULT_CONFIG_FILE "/etc/l2tp/l2tpd.conf"
-#define ALT_DEFAULT_CONFIG_FILE "/etc/l2tpd/l2tpd.conf"
+#define DEFAULT_AUTH_FILE "/etc/l2tpd/l2tp-secrets"
+#define ALT_DEFAULT_AUTH_FILE "/etc/l2tp/l2tp-secrets"
+#define DEFAULT_CONFIG_FILE "/etc/l2tpd/l2tpd.conf"
+#define ALT_DEFAULT_CONFIG_FILE "/etc/l2tp/l2tpd.conf"
 #define DEFAULT_PID_FILE "/var/run/l2tpd.pid"
 
 /* Definition of an LNS */
--- doc/l2tpd.conf.5.org	Tue Aug 13 20:23:13 2002
+++ doc/l2tpd.conf.5	Sun Nov 10 15:03:43 2002
@@ -15,7 +15,7 @@
 .TP 
 .B auth file
 Specify where to find the authentication file used to authenticate
-l2tp tunnels. The default is /etc/l2tp/l2tp\-secrets.
+l2tp tunnels. The default is /etc/l2tpd/l2tp\-secrets.
 
 .TP 
 .B Port
@@ -141,7 +141,7 @@
 
 .SH "FILES"
 
-\fB\fR/etc/l2tp/l2tpd.conf \fB\fR/etc/l2tp/l2tp\-secrets 
+\fB\fR/etc/l2tpd/l2tpd.conf \fB\fR/etc/l2tpd/l2tp\-secrets 
 \fB\fR/var/run/l2tp\-control
 .SH "BUGS"
 
--- doc/l2tpd.8.org	Mon Aug 19 17:44:45 2002
+++ doc/l2tpd.8	Sun Nov 10 15:05:25 2002
@@ -30,7 +30,7 @@
 
 .SH "FILES"
 
-\fB\fR/etc/l2tp/l2tpd.conf \fB\fR/etc/l2tp/l2tp\-secrets 
+\fB\fR/etc/l2tpd/l2tpd.conf \fB\fR/etc/l2tpd/l2tp\-secrets 
 \fB\fR/var/run/l2tp\-control
 .SH "BUGS"
 
